front/src/Phaser/Game/GameScene.ts

```diff
@@ -1044,18 +1044,16 @@ ${escapedMessage}
             })
         );
 
-        this.iframeSubscriptionList.push(
-            iframeListener.gameStateStream.subscribe(() => {
-                iframeListener.sendGameStateEvent({
-                    mapUrl: this.MapUrlFile,
-                    startLayerName: this.startPositionCalculator.startLayerName,
-                    uuid: localUserStore.getLocalUser()?.uuid,
-                    nickname: localUserStore.getName(),
-                    roomId: this.RoomId,
-                    tags: this.connection ? this.connection.getAllTags() : [],
-                });
-            })
-        );
+        iframeListener.registerAnswerer('getState', () => {
+            return Promise.resolve({
```

Comment:
PS you dont need to return a promise if you already have the value (at least not santactically - maybe we can adjust the typings to (Promise<T> |T) though)

Reply:
Holy cow, this is brilliant :+1:

Reply:
probably unrelated but may come up : 
`export type UnPromise<P> = P extends Promise<infer T> ? UnPromise<T> : P`

to get the response type of a potentially nested promise